### PR TITLE
PLAT-24224: Toggling the Recycle Bin updates its scheduled task profile

### PR DESCRIPTION
**Pull request by drorsou**

### CODE REVIEW

File: alpha/apps/kaltura/lib/kPermissionManager.php [hunk 1]
```diff
@@ -1093,6 +1098,64 @@ public function objectCreated(BaseObject $object)
...
+		elseif ($permission->getStatus() == PermissionStatus::BLOCKED)
+		{
+			self::disableRecycleBinScheduledTaskProfile($permission->getPartnerId());
+		}
+	}
+	
+	protected static function enableRecycleBinScheduledTaskProfile($partnerId)
+	{
+		$c = new Criteria();
+		$c->addAnd(ScheduledTaskProfilePeer::NAME, 'recycleBinCleanup', Criteria::EQUAL);
```
Comment (inbal-ben-david) [inline alpha/apps/kaltura/lib/kPermissionManager.php:1116]: put recycleBinCleanup in const

Comment (inbal-ben-david) [inline alpha/apps/kaltura/lib/kPermissionManager.php:1116]: use system name instead, system_name_partner_id is index on this table

File: alpha/apps/kaltura/lib/kPermissionManager.php [hunk 2]
```diff
@@ -1093,6 +1098,64 @@ public function objectCreated(BaseObject $object)
...
+		$c->addAnd(ScheduledTaskProfilePeer::PARTNER_ID, $partnerId, Criteria::EQUAL);
+		$results = ScheduledTaskProfilePeer::doSelect($c);
```
Comment (inbal-ben-david) [inline alpha/apps/kaltura/lib/kPermissionManager.php:1118]: you can just use doSelectOne

File: alpha/apps/kaltura/lib/kPermissionManager.php [hunk 3]
```diff
@@ -1093,6 +1098,64 @@ public function objectCreated(BaseObject $object)
...
+		if ($results)
+		{
+			$scheduledTaskProfile = $results[0];
+			$scheduledTaskProfile->setStatus(ScheduledTaskProfileStatus::ACTIVE);
+			$scheduledTaskProfile->save();
+		}
+		else
+		{
+			$scheduledTaskProfile = new ScheduledTaskProfile();
+			$scheduledTaskProfile->setName('recycleBinCleanup');
```
Comment (inbal-ben-david) [inline alpha/apps/kaltura/lib/kPermissionManager.php:1129]: same const, add also the system name

File: alpha/apps/kaltura/lib/kPermissionManager.php [hunk 4]
```diff
@@ -1093,6 +1098,64 @@ public function objectCreated(BaseObject $object)
...
+			$scheduledTaskProfile->setStatus(ScheduledTaskProfileStatus::ACTIVE);
+			$scheduledTaskProfile->setObjectFilterEngineType(ObjectFilterEngineType::RECYCLE_BIN_CLEANUP);
+			$scheduledTaskProfile->setPartnerId($partnerId);
+			$filter = new mediaEntryFilter;
+			$scheduledTaskProfile->setObjectFilter($filter);
+			$objectTasks = new kObjectTask();
```
Comment (inbal-ben-david) [inline alpha/apps/kaltura/lib/kPermissionManager.php:1135]: do we even need the object task? you have your own processor

File: release-notes.md [hunk 5]
```diff
@@ -1,3 +1,22 @@
+# Scorpius-19.8.0
+## Add Recycle Bin batch for auto deletion
+* Issue Type: Task
+* Issue ID: PLAT-24229
+
+### Configuration ###
+Add the following to batch.ini under KScheduledTaskRunner worker
+```
+[KAsyncRecycleBin : KScheduledTaskRunner]
```
Comment (inbal-ben-david) [inline release-notes.md:9]: you have KAsyncRecycleBin twice

File: alpha/apps/kaltura/lib/kPermissionManager.php [hunk 6]
```diff
@@ -1093,6 +1098,64 @@ public function objectCreated(BaseObject $object)
 		return true;
 	}
 	
+	protected static function handleRecycleBinPermission($permission)
```
Comment (inbal-ben-david) [inline alpha/apps/kaltura/lib/kPermissionManager.php:1101]: worth considering storing the max days on this profile as well and not on partner level

File: alpha/apps/kaltura/lib/kPermissionManager.php [hunk 7]
```diff
@@ -1093,6 +1098,64 @@ public function objectCreated(BaseObject $object)
...
+			$objectTasks->setStopProcessingOnError(true);
+			$scheduledTaskProfile->setObjectTasks($objectTasks);
+			$scheduledTaskProfile->setMaxTotalCountAllowed(50);
+			$scheduledTaskProfile->save();
+		}
+	}
+	
+	protected static function disableRecycleBinScheduledTaskProfile($partnerId)
+	{
+		$c = new Criteria();
```
Comment (inbal-ben-david) [inline alpha/apps/kaltura/lib/kPermissionManager.php:1146]: can be putted in getRecycleBinScheduledTaskProfile function

File: alpha/apps/kaltura/lib/kPermissionManager.php [hunk 8]
```diff
@@ -1056,7 +1060,7 @@ public function objectChanged(BaseObject $object, array $modifiedColumns)
 	 */
 	public function shouldConsumeCreatedEvent(BaseObject $object)
 	{
-		if($object instanceof Permission)
+		if ($object instanceof Permission && $object->getPartnerId() != PartnerPeer::GLOBAL_PARTNER)
```
[the remark this answers is not in the document]
Comment (drorsou) [reply alpha/apps/kaltura/lib/kPermissionManager.php:1063]: I added this condition because the objectCreated handling already had the same condition.

**Review by inbal-ben-david — APPROVED** [review]

**Pull request merged**